src/domain/course/CoursePageContainer.tsx

```diff
@@ -39,8 +46,39 @@ const CoursePageContainer: React.FC = () => {
     },
   });
 
+  const [getData, { data: superEvent }] = useCourseDetailsLazyQuery({
+    variables: {
+      id: superEventId,
+      include: ['in_language', 'keywords', 'location', 'audience'],
+    },
+  });
+
   const course = courseData?.courseDetails;
 
+  useEffect(() => {
+    if (course) {
+      const superEventIdData = course.superEvent?.internalId
+        ?.split('/')
+        .filter((e) => e)
+        .pop();
+      if (superEventIdData) {
+        setSuperEventId(superEventIdData);
+      } else {
+        setSuperEventData(undefined);
+      }
+    }
+  }, [course]);
+
+  useEffect(() => {
+    if (superEventId) {
+      getData();
+    }
+  }, [getData, superEventId]);
+
+  useEffect(() => {
+    setSuperEventData(superEvent?.courseDetails || null);
+  }, [superEvent]);
```

Comment:
Do we need to split this into 3 useEffects? Could this be handled in one?

Reply:
i think 3 needed. we can call getData only when we have superEventId, so we have to store it in state, as we get it from the first query. superEventData can be undefined (if no superevent id), null (when we dont know if there is superevent id, or when its loading) and the actual superEvent object. To handle all those 3, and not make EventHero blinking with wrong data, i think we have to keep logic like that. Effects just allow to make a sync sequence. But if you have better implementation, please tell:)

Reply:
There is some discussion about it: https://stackoverflow.com/questions/54002792/should-i-use-one-or-many-useeffect-in-component
I think first two useEffects fall to category 2 and should be separated.
But can you merge last two useEffects like this?
```
  useEffect(() => {
    if (superEventId) {
      const superEventData = getData();
      setSuperEventData(superEventData);
    }
  }, [getData, superEventId]);
  ```

Reply:
Also I would still put them all to custom hook, it's more readable and more reusable (if needed elsewhere later)

Reply:
If we really need to have separate state for superEventData. I wouldn't use lazy hook in the first place.

Reply:
what is the solution without lazy hook and why it is bad?